diff --git a/res/menu/options_menu.xml b/res/menu/options_menu.xml
index 3efaeb7..0a0ac9e 100644
--- a/res/menu/options_menu.xml
+++ b/res/menu/options_menu.xml
@@ -17,12 +17,11 @@
 -->
 <!-- Options Menu for SearchableActivity and EntryActivity.
  -->
+<!-- BACKPORT: No action view. -->
 <menu xmlns:android="http://schemas.android.com/apk/res/android">
     <item android:id="@+id/search"
           android:title="@string/menu_search"
-          android:icon="@drawable/ic_menu_search"
-          android:showAsAction="ifRoom|collapseActionView"
-          android:actionViewClass="android.widget.SearchView" />
+          android:icon="@drawable/ic_menu_search" />
 
     <item android:id="@+id/about"
         android:title="@string/about" />
diff --git a/src/org/tlhInganHol/android/klingonassistant/EntryActivity.java b/src/org/tlhInganHol/android/klingonassistant/EntryActivity.java
index 1bc0a9d..46db6d7 100644
--- a/src/org/tlhInganHol/android/klingonassistant/EntryActivity.java
+++ b/src/org/tlhInganHol/android/klingonassistant/EntryActivity.java
@@ -18,16 +18,13 @@ package org.tlhInganHol.android.klingonassistant;
 
 import java.util.regex.Matcher;
 
-import android.app.ActionBar;
 import android.app.Activity;
 import android.app.SearchManager;
 import android.content.ComponentName;
-import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
 import android.database.Cursor;
 import android.net.Uri;
-import android.os.Build;
 import android.os.Bundle;
 import android.preference.PreferenceManager;
 import android.text.Html;
@@ -44,8 +41,7 @@ import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.view.View;
-import android.widget.SearchView;
 import android.widget.TextView;
 
 /**
  * Displays an entry and its definition.
@@ -67,13 +63,10 @@ public class EntryActivity extends Activity {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.entry);
 
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB){
-            ActionBar actionBar = getActionBar();
-            actionBar.setDisplayHomeAsUpEnabled(true);
-        }
+        // BACKPORT: No Action Bar.
 
         TextView entryTitle = (TextView) findViewById(R.id.entry);
         TextView entryText = (TextView) findViewById(R.id.definition);

         // TODO: Save and restore bundle state to preserve links.
 
@@ -231,19 +224,19 @@ public class EntryActivity extends Activity {
         }
         if (!definition_DE.equals("")) {
             // Reduce the size of the German definition.
-            ssb.setSpan(new AbsoluteSizeSpan(14, true), germanDefinitionStart,
+            ssb.setSpan(new AbsoluteSizeSpan(14), germanDefinitionStart,
                 germanDefinitionStart + germanDefinitionHeader.length() +
                 definition_DE.length(), finalFlags);
         }
         if (showTransitivityInformation) {
             // Reduce the size of the transitivity information.
-            ssb.setSpan(new AbsoluteSizeSpan(14, true), transitivityStart,
+            ssb.setSpan(new AbsoluteSizeSpan(14), transitivityStart,
                 transitivityStart + transitivityHeader.length() +
                 transitivity.length(), finalFlags);
         }
         if (!hiddenNotes.equals("")) {
             // Reduce the size of the hidden notes.
-            ssb.setSpan(new AbsoluteSizeSpan(14, true), hiddenNotesStart,
+            ssb.setSpan(new AbsoluteSizeSpan(14), hiddenNotesStart,
                 hiddenNotesStart + hiddenNotesHeader.length() +
                 hiddenNotes.length(), finalFlags);
         }
@@ -285,7 +278,7 @@ public class EntryActivity extends Activity {
             // If linked entry is hypothetical or extended canon, insert a "?" in front.
             if (linkedEntry.isHypothetical() || linkedEntry.isExtendedCanon()) {
                 ssb.insert(m.start(), "?");
-                ssb.setSpan(new AbsoluteSizeSpan(14, true), m.start(), m.start() + 1, intermediateFlags);
+                ssb.setSpan(new AbsoluteSizeSpan(14), m.start(), m.start() + 1, intermediateFlags);
                 ssb.setSpan(new SuperscriptSpan(), m.start(), m.start() + 1, maybeFinalFlags);
                 end++;
             }
@@ -334,12 +327,7 @@ public class EntryActivity extends Activity {
         MenuInflater inflater = getMenuInflater();
         inflater.inflate(R.menu.options_menu, menu);
 
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB){
-            SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
-            SearchView searchView = (SearchView) menu.findItem(R.id.search).getActionView();
-            searchView.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
-            searchView.setIconifiedByDefault(false);
-        }
+        // BACKPORT: No search view.
 
         return true;
     }
@@ -350,11 +338,7 @@ public class EntryActivity extends Activity {
             case R.id.search:
                 onSearchRequested();
                 return true;
-            case android.R.id.home:
-                Intent intent = new Intent(this, KlingonAssistant.class);
-                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-                startActivity(intent);
-                return true;
+            // BACKPORT: No handling of android.R.id.home.
             case R.id.about:
                 // Show "About" screen.
                 displayHelp(QUERY_FOR_ABOUT);
diff --git a/src/org/tlhInganHol/android/klingonassistant/KlingonAssistant.java b/src/org/tlhInganHol/android/klingonassistant/KlingonAssistant.java
index b26ef06..6bcd1c4 100644
--- a/src/org/tlhInganHol/android/klingonassistant/KlingonAssistant.java
+++ b/src/org/tlhInganHol/android/klingonassistant/KlingonAssistant.java
@@ -16,7 +16,6 @@
 
 package org.tlhInganHol.android.klingonassistant;
 
-import android.app.ActionBar;
 import android.app.Activity;
 import android.app.SearchManager;
 import android.content.Context;
@@ -25,7 +24,6 @@ import android.content.SharedPreferences;
 import android.database.Cursor;
 import android.graphics.Typeface;
 import android.net.Uri;
-import android.os.Build;
 import android.os.Bundle;
 import android.preference.PreferenceManager;
 import android.text.Html;
@@ -39,7 +37,6 @@ import android.view.ViewGroup;
 import android.widget.AdapterView;
 import android.widget.BaseAdapter;
 import android.widget.ListView;
-import android.widget.SearchView;
 import android.widget.TextView;
 import android.widget.TwoLineListItem;
 
@@ -71,11 +68,6 @@ public class KlingonAssistant extends Activity {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.main);
 
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB){
-            ActionBar actionBar = getActionBar();
-            actionBar.setDisplayHomeAsUpEnabled(true);
-        }
-
         mTextView = (TextView) findViewById(R.id.text);
         mListView = (ListView) findViewById(R.id.list);
 
@@ -246,12 +238,7 @@ public class KlingonAssistant extends Activity {
         MenuInflater inflater = getMenuInflater();
         inflater.inflate(R.menu.options_menu, menu);
 
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB){
-            SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
-            SearchView searchView = (SearchView) menu.findItem(R.id.search).getActionView();
-            searchView.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
-            searchView.setIconifiedByDefault(false);
-        }
+        // BACKPORT: No search view.
 
         return true;
     }
@@ -262,9 +249,7 @@ public class KlingonAssistant extends Activity {
             case R.id.search:
                 onSearchRequested();
                 return true;
-            case android.R.id.home:
-                finish();
-                return true;
+            // BACKPORT: No handling of android.R.id.home.
             case R.id.about:
                 // Show "About" screen.
                 showResults(QUERY_FOR_ABOUT);
diff --git a/src/org/tlhInganHol/android/klingonassistant/Preferences.java b/src/org/tlhInganHol/android/klingonassistant/Preferences.java
index dad98a5..a5f44b9 100644
--- a/src/org/tlhInganHol/android/klingonassistant/Preferences.java
+++ b/src/org/tlhInganHol/android/klingonassistant/Preferences.java
@@ -18,7 +18,6 @@ package org.tlhInganHol.android.klingonassistant;
 
 import org.tlhInganHol.android.klingonassistant.R;
 
-import android.app.ActionBar;
 import android.content.Intent;
 import android.os.Build;
 import android.os.Bundle;
@@ -47,18 +46,13 @@ public class Preferences extends PreferenceActivity {
         // Load the preferences from an XML resource.
         addPreferencesFromResource(R.xml.preferences);
 
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB){
-            ActionBar actionBar = getActionBar();
-            actionBar.setDisplayHomeAsUpEnabled(true);
-        }
+        // BACKPORT: No Action Bar.
     }
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
-            case android.R.id.home:
-                finish();
-                return true;
+            // BACKPORT: No handling of android.R.id.home.
             default:
                 return super.onOptionsItemSelected(item);
         }
